src/main/kotlin/org/opensearch/commons/alerting/model/Alert.kt

```diff
@@ -0,0 +1,373 @@
+package org.opensearch.commons.alerting.model
+
+import org.opensearch.common.io.stream.StreamInput
+import org.opensearch.common.io.stream.StreamOutput
+import org.opensearch.common.io.stream.Writeable
+import org.opensearch.common.lucene.uid.Versions
+import org.opensearch.common.xcontent.ToXContent
+import org.opensearch.common.xcontent.XContentBuilder
+import org.opensearch.common.xcontent.XContentParser
+import org.opensearch.common.xcontent.XContentParserUtils.ensureExpectedToken
+import org.opensearch.commons.alerting.alerts.AlertError
+import org.opensearch.commons.alerting.util.IndexUtils.Companion.NO_SCHEMA_VERSION
+import org.opensearch.commons.alerting.util.instant
+import org.opensearch.commons.alerting.util.optionalTimeField
+import org.opensearch.commons.alerting.util.optionalUserField
+import org.opensearch.commons.authuser.User
+import java.io.IOException
+import java.time.Instant
+
+data class Alert(
+    val id: String = NO_ID,
+    val version: Long = NO_VERSION,
+    val schemaVersion: Int = NO_SCHEMA_VERSION,
+    val monitorId: String,
+    val monitorName: String,
+    val monitorVersion: Long,
+    val monitorUser: User?,
+    val triggerId: String,
+    val triggerName: String,
+    val findingIds: List<String>,
+    val relatedDocIds: List<String>,
+    val state: State,
+    val startTime: Instant,
+    val endTime: Instant? = null,
+    val lastNotificationTime: Instant? = null,
+    val acknowledgedTime: Instant? = null,
+    val errorMessage: String? = null,
+    val errorHistory: List<AlertError>,
+    val severity: String,
+    val actionExecutionResults: List<ActionExecutionResult>,
+    val aggregationResultBucket: AggregationResultBucket? = null
+) : Writeable, ToXContent {
```

Comment:
Is there a room to segregate the PoJo for Alert model required for response vs one required for data persistence and transport.